components/servo/webview.rs

```diff
@@ -169,6 +174,32 @@ impl WebView {
         webview
     }
 
+    pub fn create_entangled_message_ports(&self) -> (MessagePort, MessagePort) {
+        let constellation_proxy = self.inner().constellation_proxy.clone();
+        let embedder_proxy = self.inner().embedder_proxy.clone();
+
+        let (port1, port2) =
+            MessagePort::new_entangled(constellation_proxy, embedder_proxy, self.id());
+        self.inner_mut()
+            .message_ports
+            .extend([port1.clone(), port2.clone()]);
+        (port1, port2)
+    }
```

Comment:
Not sure about this API. 

Could you not just return an entangled port, and ship the other to the pipeline by default? In JS script can do all kinds of things with ports, including just using them in the same global, but that doesn't seem useful for the embedder.